3_javascript/javascript1/week3/script.js

```diff
@@ -0,0 +1,217 @@
+
+function travelInformation(speedInkmperh, destinationDistanceInkm) {
+
+    let timeToRichToTheDestination = (destinationDistanceInkm / speedInkmperh);
+    const hours = Math.trunc(timeToRichToTheDestination); //same result with const hours (~~timeToRichToTheDestination) to cuts all fractional digits
+    let minutes = (timeToRichToTheDestination - hours) * 60;
+    const rminutes = Math.ceil(minutes); // To round up the minutes
+    return ('Your travel takes ${hours} hours & ${rminutes} minutes'); // It doesn't give a result for ${hours} & ${rminutes} !!!!!
+};
+
+const travelTime = travelInformation(50, 432);
+console.log(travelTime);
+
+
+// ______ warmup excercise : how much time a tv series have taken as a percentage of an average lifespan of 80 years_______
+
+function logOutSeriesText(TVSeriestiltle, daysToWatch) {
+
+    const tookDaysOfLife = (80 * 365) / daysToWatch;
+    let percentFromlife = (daysToWatch * 100) / tookDaysOfLife;
+    let rPercentOfLife = percentFromlife.toPrecision(4); // To have just 4 digits and only 3 digits after decimal point
+    return (`${TVSeriestiltle} took ${rPercentOfLife} % of my life`);
+};
+
+const seriesDurations = [
+    {
+        title: "Game of thrones",
+        days: 9,
+        hours: 1,
+        minutes: 0,
+    },
+    {
+        title: "Sopranos",
+        days: 19,
+        hours: 14,
+        minutes: 0,
+    },
+    {
+        title: "The Wire",
+        days: 27,
+        hours: 12,
+        minutes: 0,
+    },
+];
+
+
+for (let i = 0; i < seriesDurations.length; i++) {
+    let result = logOutSeriesText(seriesDurations[i].title, seriesDurations[i].days);
+    console.log(result);
+    // Could you please tell me how I can add the 3 percentages and make a subtotal of them? :), Thank you.
+};
+
+
+
+// _________Homework : Smart-ease / NOnoN0nOYes (Note taking app) / create a simple and easy notepad__________
+
+// ---------Save a note  ---------------------------------
+
+const notes = [];
+
+function saveNote(content, id) { // content parameter is a string and id is a number
+    let note = { content: content, id: id };
+    notes.push(note);
+    return notes;
+};
+
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+saveNote("Washing dishes", 3);
+saveNote("Ironing clothes", 4);
+console.log(notes); // [{content: 'Pick up groceries', id: 1}, {content: 'Do laundry', id: 2}, ........]
+
+
+//------  Get a note  ----------------------------------
+
+function getNote(id) {
+
+    if (id <= notes.length || typeof id !== 'string') {
+        let noteT = notes.find(item => item.id == id);
+        return (noteT.content); //f.x  id=1 => "Pick up groceries"
+    } else {
+
+        return (alert(`Are you kidding me? :O A number !!
+        Run the program again and enter a number between 1 and ${notes.length}`));
+        //Why I get 'Undefind' after this alert message? :/
+    }
+};
+
+const getId = prompt(`Enter a number between 1 and ${notes.length} `, 1);
```

Comment:
Thank you :)
Could you please tell me why the codes between lines 78 to 95 do not work properly? 
And why do I get 'Undefined' after this alert message?

Reply:
yes :)  if (typeof id === `number`) issue is here, if you want to refer to a type, you must to use quotes :) Otherwise it will look for a variable `number` (e.g const number) but there isn't such a variable so it throws an error :)

Reply:
Also, you must first parse the ﻿`id` to a number because all input values from the prompt come as a ﻿`string`.
To do so, you can either use casting as: ﻿`Number(id),` or simply add a ﻿`+` before it like: ﻿`+id`. Both approaches are valid. This only works on numbers within strings. If you try to give the ﻿id a value of "my-id", it won't work. That scenario would be caught by the else case, which is the correct behavior.